### PR TITLE
fix 'transaction.timeout.ms' not used warning

### PR DESCRIPTION
**Pull request by dfakhritdinov**

### CODE REVIEW

**Comment by dfakhritdinov** [comment]
as side bonus configuration 

`partitionerClass.map { value => (C.PARTITIONER_CLASS_CONFIG, value) }`

will **not** be ignored

File: skafka/src/main/scala/com/evolutiongaming/skafka/producer/ProducerConfig.scala [hunk 1]
```diff
@@ -46,12 +46,16 @@ final case class ProducerConfig(
       (C.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, maxInFlightRequestsPerConnection.toString),
       (C.INTERCEPTOR_CLASSES_CONFIG, interceptorClasses mkString ","),
       (C.ENABLE_IDEMPOTENCE_CONFIG, idempotence.toString),
-      (C.TRANSACTION_TIMEOUT_CONFIG, transactionTimeout.toMillis.toString)
     )
 
-    val bindings = bindings1 ++
-      transactionalId.map { (C.TRANSACTIONAL_ID_CONFIG, _) }
-    partitionerClass.map { value => (C.PARTITIONER_CLASS_CONFIG, value) }
+    val transactions = transactionalId.toList.flatMap { id =>
```
Comment (t3hnar) [inline skafka/src/main/scala/com/evolutiongaming/skafka/producer/ProducerConfig.scala:51]: just curious, does this work without `.toList` ?

File: skafka/src/test/scala/com/evolutiongaming/skafka/producer/ProducerConfigSpec.scala [hunk 2]
```diff
@@ -63,7 +63,6 @@ class ProducerConfigSpec extends AnyFunSuite with Matchers {
       "max.block.ms"                          -> "60000",
       "client.id"                             -> "clientId",
       "metric.reporters"                      -> "",
-      "transaction.timeout.ms"                -> "60000",
```
Comment (t3hnar) [inline skafka/src/test/scala/com/evolutiongaming/skafka/producer/ProducerConfigSpec.scala:66]: let's better add transactionId :)

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: skafka/src/main/scala/com/evolutiongaming/skafka/producer/ProducerConfig.scala (hunk 1 above)
Comment (dfakhritdinov) [reply skafka/src/main/scala/com/evolutiongaming/skafka/producer/ProducerConfig.scala:51]: No, `Option.flatMap` expects only `Option`

File: skafka/src/test/scala/com/evolutiongaming/skafka/producer/ProducerConfigSpec.scala (hunk 2 above)
Comment (dfakhritdinov) [reply skafka/src/test/scala/com/evolutiongaming/skafka/producer/ProducerConfigSpec.scala:66]: done

**Review by t3hnar — APPROVED** [review]

**Pull request merged**